src/urllib3/connection.py

```diff
@@ -29,6 +29,7 @@
 
 from .util.timeout import _DEFAULT_TIMEOUT, _TYPE_TIMEOUT, Timeout
 from .util.util import to_str
+from .response import HTTPResponse
```

Comment:
This will need to be fixed as well, it caused a circular import. 

```
ImportError: cannot import name '_TYPE_BODY' from partially initialized module 'flit_core.dummy.import1.connection' (most likely due to a circular import) (/home/runner/work/urllib3/urllib3/src/urllib3/connection.py)
```

Reply:
This has been harder to fix than expected, lots of imports circular imports between connections.py and respone.py

Reply:
You'll need to use relative imports for urllib3, so it'd be `from .connection import X`. Reason we do this is some projects vendor us so we need to work in those circumstances too.

Reply:
okay, but the only way forward I can see right now, is to combine `connection.py` and `response.py`. Since they import so many things from each other.

Reply:
I found a way but its messy, lets come back to this.